Directory.Build.targets

```diff
@@ -1,50 +1,61 @@
 <Project>
 
-  <PropertyGroup>
-    <ExtensionsVersion>6.0.0</ExtensionsVersion>
-    <MicrosoftBuildVersion>17.3.2</MicrosoftBuildVersion>
-    <RoslynVersion>4.2.0</RoslynVersion>
-  </PropertyGroup>
-
   <ItemGroup>
-    <PackageReference Update="Google.Protobuf" Version="3.21.8" />
-    <PackageReference Update="Grpc.Tools" Version="2.50.0" />
-
+    <PackageReference Update="Google.Protobuf" Version="3.21.11" />
+    <PackageReference Update="Grpc.Tools" Version="2.51.0" />
     <PackageReference Update="CommandLineParser" Version="2.9.1" />
+    <PackageReference Update="Microsoft.SourceLink.GitHub" Version="1.1.1" />
+  </ItemGroup>
 
-    <PackageReference Update="LeanCode.Time" Version="6.0.1327" />
-
-    <PackageReference Update="Microsoft.Build" Version="$(MicrosoftBuildVersion)" />
-    <PackageReference Update="Microsoft.Build.Framework" Version="$(MicrosoftBuildVersion)" />
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net6.0' ">
+    <PackageReference Update="Microsoft.Build" Version="17.3.2" />
+    <PackageReference Update="Microsoft.Build.Framework" Version="17.3.2" />
     <PackageReference Update="Microsoft.Build.Locator" Version="1.5.5" />
 
-    <PackageReference Update="Microsoft.CodeAnalysis.CSharp" Version="$(RoslynVersion)" />
-    <PackageReference Update="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="$(RoslynVersion)" />
-    <PackageReference Update="Microsoft.CodeAnalysis.Workspaces.MSBuild" Version="$(RoslynVersion)" />
+    <PackageReference Update="Microsoft.CodeAnalysis.CSharp" Version="4.2.0" />
+    <PackageReference Update="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.2.0" />
+    <PackageReference Update="Microsoft.CodeAnalysis.Workspaces.MSBuild" Version="4.2.0" />
 
-    <PackageReference Update="Microsoft.Extensions.DependencyModel" Version="$(ExtensionsVersion)" />
-    <PackageReference Update="Microsoft.Extensions.FileSystemGlobbing" Version="$(ExtensionsVersion)" />
+    <PackageReference Update="Microsoft.Extensions.DependencyModel" Version="6.0.0" />
+    <PackageReference Update="Microsoft.Extensions.FileSystemGlobbing" Version="6.0.0" />
+    <PackageReference Update="System.Reflection.MetadataLoadContext" Version="6.0.0" />
+  </ItemGroup>
 
-    <PackageReference Update="Microsoft.SourceLink.GitHub" Version="1.1.1" />
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net7.0' ">
```

Comment:
Can we leave versions in a separate `PropertyGroup` instead? E.g.

```xml
  <PropertyGroup>
    <ExtensionsVersion>7.0.0</ExtensionsVersion>
    <MicrosoftBuildVersion>17.4.0</MicrosoftBuildVersion>
  </PropertyGroup>

  <PropertyGroup Condition="'$(TargetFramework)' == 'net6.0'">
    <RoslynVersion>4.2.0</RoslynVersion>
  </PropertyGroup>

  <PropertyGroup Condition="'$(TargetFramework)' == 'net7.0'">
    <RoslynVersion>4.4.0</RoslynVersion>
  </PropertyGroup>
```

And while we're at it, is there anything preventing us from using 7.0.0 extensions and 17.4 MSBuild with `net6.0` target?

Reply:
To be honest, using conditional `ItemGroup`s with all the versions specified explicitly is, IMO, more readable.

> And while we're at it, is there anything preventing us from using 7.0.0 extensions and 17.4 MSBuild with `net6.0` target?

reg. extensions - I think we can use them on both. Reg. MSBuild - I would really like to keep that consistent with what the SDK ships, so that we have a consistent behavior (at least I hope so).

Reply:
Doesn't latest 6.0 SDK ship MSBuild 17.4 already?

Reply:
Nope:
![image](https://user-images.githubusercontent.com/357096/207057033-136eb659-113b-4d81-b4c2-5f9492d825fd.png)

Reply:
Ah yeah, I just noticed that myself:

```
$ dotnet /usr/share/dotnet/sdk/6.0.403/MSBuild.dll --version
MSBuild version 17.3.2+561848881 for .NET
17.3.2.46306
```

Makes sense to keep it at 17.3.2 in that case.